solidity/ecdsa/contracts/libraries/Wallets.sol

```diff
@@ -14,22 +14,19 @@
 
 pragma solidity ^0.8.9;
 
-import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
 // TODO: This contract is just a Stub implementation that was used for gas
 // comparisons for Wallets creation. It should be implemented according to the
 // Wallets' actual use case.
 library Wallets {
-    using ECDSA for bytes32;
-
     struct Wallet {
-        // TODO: Verify if we want to store the whole public key or having
-        // just the public key hash is enough.
         // Keccak256 hash of group members identifiers array. Group members do not
         // include operators selected by the sortition pool that misbehaved during DKG.
         bytes32 membersIdsHash;
-        bytes32 digestToSign;
+        // Uncompressed ECDSA public key (64-byte long) as a concatenation of X
+        // and Y coordinates.
+        bytes publicKey;
```

Comment:
> We should store the key in a compressed form 

We need both compressed and uncompressed public keys format in the Bridge to validate transactions and signatures. It's easier to convert the key from uncompressed to compressed format than the other way around. As it would require elliptic curve operations.

> use the remaining space for uint64 wallet creation date. This will be needed to manage the wallet lifecycle. Please look at wallet_creation_period and wallet_max_age parameters in [TBTCv2 RFC-1](https://github.com/keep-network/tbtc-v2/blob/main/docs/rfc/rfc-1.adoc#26-wallet-lifecycle).

We probably will want to store it in the Bridge next to the other wallet details as the Bridge will manage the wallet lifecycle as the owner.

> Have you tried to replace it with two bytes32 slots

Nice catch, it reduced cost for about 24k (fe77e91f7f5a5f81a0bc199fdc989810eb08970f)

Reply:
Is the gas consumption in the PR description up-to-date?

Reply:
It is now.